runtime/src/main/java/app/cash/paraphrase/DateTimeConverter.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2023 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.paraphrase
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+
+/**
+ * Converts `java.time` types used by Paraphrase to a [Calendar] that can be used by ICU to format.
+ *
+ * [Calendar] is generic so the system-appropriate ICU calendar implementation can be used:
+ * `android.icu.util` on Android, or `com.ibm.icu` on the JVM.
+ */
+public interface DateTimeConverter<out Calendar : Any> {
```

Comment:
Possibly I should add a "not safe for extension" disclaimer? I'd make it `sealed` but the JVM implementation needs to be in a different module.

Reply:
You want https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-subclass-opt-in-required/ plus an annotation set to error.

Reply:
Cool, done